src/main/java/org/ow2/proactive/addons/ldap_query/LDAPClient.java

```diff
@@ -0,0 +1,232 @@
+import org.ow2.proactive.addons.ldap_query.model.LDAPResponse;
+import org.ow2.proactive.addons.ldap_query.model.Response;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+
+@Setter
+@NoArgsConstructor
+public class LDAPClient {
+
+    private static final String REGEX_LIST_SEPARATOR = ",\\s?";
+
+    /*
+     * Define name of variables that can be passed to the LDAP Query task
+     */
+
+    public static final String ARG_URL = "ldapUrl";
+
+    public static final String ARG_USERNAME = "ldapUsername";
+
+    public static final String ARG_PASSWORD = "ldapPassword";
+
+    public static final String ARG_SEARCH_BASE = "ldapSearchBase";
+
+    public static final String ARG_SEARCH_FILTER = "ldapSearchFilter";
+
+    public static final String ARG_SELECTED_ATTRIBUTES = "ldapSelectedAttributes";
+
+    protected String ldapUrl;
+
+    protected String ldapUsername;
+
+    protected String ldapPassword;
+
+    protected String ldapSearchBase;
+
+    protected String ldapSearchFilter;
+
+    protected String ldapSelectedAttributes;
+
+    protected DirContext ldapConnection;
+
+    public LDAPClient(String ldapUrl, String ldapUsername, String ldapPassword, String ldapSearchBase,
+            String ldapSearchFilter, String ldapSelectedAttributes) {
+        this.ldapUrl = ldapUrl;
+        this.ldapUsername = ldapUsername;
+        this.ldapPassword = ldapPassword;
+        this.ldapSearchBase = ldapSearchBase;
+        this.ldapSearchFilter = ldapSearchFilter;
+        this.ldapSelectedAttributes = ldapSelectedAttributes;
+    }
+
+    public LDAPClient(Map<String, Serializable> actualTaskVariables, Map<String, Serializable> credentials) {
+        List<String> taskVariablesList = new LinkedList<>();
+        taskVariablesList.add(ARG_URL);
+        taskVariablesList.add(ARG_USERNAME);
+        taskVariablesList.add(ARG_PASSWORD);
+        taskVariablesList.add(ARG_SEARCH_BASE);
+        taskVariablesList.add(ARG_SEARCH_FILTER);
+        taskVariablesList.add(ARG_SELECTED_ATTRIBUTES);
+
+        Map mapWithVariables;
+        for (String variableName : taskVariablesList) {
+            if (actualTaskVariables.containsKey(variableName)) {
+                mapWithVariables = actualTaskVariables;
+            } else {
+                mapWithVariables = credentials;
+            }
+            setLdapClientFields(mapWithVariables, variableName);
+        }
+    }
+
+    private void setLdapClientFields(Map<String, Serializable> taskVariablesMap, String variableName) {
+        if (!taskVariablesMap.containsKey(variableName)) {
+            throw new IllegalArgumentException("The missed argument for LDAPClient, variable name: " + variableName);
+        }
+        String varValue = getAsString(taskVariablesMap, variableName);
+        try {
+            this.getClass().getDeclaredField(variableName).set(this, varValue);
```

Comment:
I don't like introspection for such a use case. It's easy to break in the future and it has impacts on performance.

You have checked just before that the expected variable exists. Why do you need to copy fields to dedicated fields. Is it not possible to simply query the map when required?